src/main/java/com/edmunds/rest/databricks/DatabricksServiceFactory.java

```diff
@@ -144,4 +154,85 @@ public DbfsService getDbfsService() {
     }
     return dbfsService;
   }
+
+  /**
+   * This is how the DatabricksServiceFactory should be constructed. This gives flexibility to add
+   * more parameters later without ending up with large constructors.
+   */
+  public static class Builder {
+
+    long retryInterval = DEFAULT_HTTP_CLIENT_RETRY_INTERVAL;
+    int maxRetries = DEFAULT_HTTP_CLIENT_MAX_RETRY;
+    String token;
+    String host;
+    String username;
+    String password;
+
+    private Builder() {
+      //NO-OP
+    }
+
+    /**
+     * Creates a DatabricksServiceFactory using token authentication.
+     *
+     * @param token your databricks token
+     * @param host the databricks host where that token is valid
+     * @return the builder object
+     */
+    public static Builder createServiceFactoryWithTokenAuthentication(String token, String host) {
+      Builder builder = new Builder();
+      builder.token = token;
+      builder.host = host;
+      return builder;
+    }
+
+    /**
+     * Creates a DatabrickServiceFactory using username password authentication.
+     *
+     * @param username databricks username
+     * @param password databricks password
+     * @param host the host object
+     * @return the builder object
+     */
+    public static Builder createServiceFactoryWithUserPasswordAuthentication(String username,
+        String password, String host) {
+      Builder builder = new Builder();
+      builder.username = username;
+      builder.password = password;
+      builder.host = host;
+      return builder;
+    }
+
+    public Builder withMaxRetries(int maxRetries) {
+      this.maxRetries = maxRetries;
+      return this;
+    }
+
+    public Builder withRetryInterval(long retryInterval) {
+      this.retryInterval = retryInterval;
+      return this;
+    }
+
+    /**
+     * Builds a DatabricksServiceFactory. Conducts basic validation.
+     *
+     * @return the databricks service factory object
+     */
+    public DatabricksServiceFactory build() {
+      if (token != null) {
```

Comment:
Do you think null check is sufficient, or, should it also check blank?

Reply:
added a comment that this was not intended to be exhaustive validation, and is instead just determining which client to build.

Also added blank checks.